src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -447,6 +435,140 @@ private void sortAvailableModules(List<Module> modulesAvailable, UserProfile use
+                || m.toString().startsWith("GES")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.UNIVERSITY_LEVEL_REQUIREMENTS, new int[] {count});
+    }
+
+    /**
+     * Insert the number of foundation modules to the credit mapping.
+     */
+    private void mapFoundation() {
+        int numOfFoundation = countProgrammeRequirementNonBd(DegreeRequirement.FOUNDATION);
+        statusMap.put(DegreeRequirement.FOUNDATION, new int[] {numOfFoundation});
+    }
+
+    /**
+     * Insert the number of mathematics modules to the credit mapping.
+     */
+    private void mapMathematics() {
+        int numOfMathematics = countProgrammeRequirementNonBd(DegreeRequirement.MATHEMATICS);
+        statusMap.put(DegreeRequirement.MATHEMATICS, new int[] {numOfMathematics});
+    }
+
+    /**
+     * Insert the number of science modules to the credit mapping.
+     */
+    private void mapScience() {
+        int numOfScience = countProgrammeRequirementNonBd(DegreeRequirement.SCIENCE);
+        statusMap.put(DegreeRequirement.SCIENCE, new int[] {numOfScience});
+    }
+
+    /**
+     * Insert the number of IT professionalism modules to the credit mapping..
+     */
+    private void mapItProfessionalism() {
+        int numOfItProfessionalism = countProgrammeRequirementNonBd(
+                                        DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT);
+        statusMap.put(DegreeRequirement.IT_PROFESSIONALISM, new int[] {numOfItProfessionalism});
+    }
+
+    /**
+     * Insert the number of industrial experience requirement modules to the credit mapping.
+     */
+    private void mapIndustrialExperienceRequirement() {
+        int numOfIndExpReq = countProgrammeRequirementNonBd(DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT);
+        statusMap.put(DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT, new int[] {numOfIndExpReq});
+    }
+
+    /**
+     * Count the number of Team Project Modules and insert to the credit mapping.
+     */
+    private void mapTeamProject() {
+        int numOfTeamProject = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement().equals(DegreeRequirement.BREATH_AND_DEPTH)
+                    && moduleDescription.get().getFocusAreas().isEmpty()) {
+                numOfTeamProject += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.TEAM_PROJECT, new int[] {numOfTeamProject});
+    }
+
+    /**
+     * Count the number of the user's focus area requirement and insert to the credit mapping.
+     */
+    private void mapFocusAreasRequirement() {
+        List<FocusArea> focusAreas = new ArrayList<>(getUserProfile().getFocusAreas());
+        if (!focusAreas.isEmpty()) {
```

Comment:
Can convert the `else` of this part into a guard clause

Reply:
The arrow head is a little deep. Best to refactor

Reply:
Im not sure what you mean, but there's no 'else' ?  I think the method's not too long

Reply:
Hmm ok ignore what I said on the else.
It's not that long, but the level of nesting is a little deep. Total got 4 levels. But ah should be fine

Reply:
It looks deeper because of the optional haha